### ISSUE TITLE
Training DINO

### ISSUE THREAD

**Issue by cmartin-isla** [lucidrains/vit-pytorch#154]
Hello, I am trying to train DINO with a base ViT from scratch and I have some doubts.
First of all, I think that in the original paper, the student temperature is 0.1 in the 30 epoch warmup, but I am not able to find the default value of 0.9 of this repo in the original paper.

 After adding cosine scheduler and doing some tweaking, I manage to have a loss of the 1e-7, but this loss has an oscillation., **every 3-4 epochs it reach a maximum around 20-30 and then decreases another time to new minima near the 1e-7 order.** I don't know if you experienced that kind of behavior.
 Secondly, after training, I want to visualize the attention maps but I am not sure how to do that. Let's say that I have trained with  224x244 images and patch size of 8. I have attention maps of (1, 6, 8, 785, 785), so 28*28 patches + 1 cls token per patch = 785:

-  In which order should I rearrange such patches, let's say the CLS tokens? 
-  If the attention is not reciprocal, should I take the last column or the last row?
- In the DINO paper, it seems that the attention maps are not of a reduced resolution, do you think they are interpolating them up?

Thanks a lot, and again, thanks for this amazing repo.

**Comment by adeschemps** [comment]
I've had the exact same issue, even without cosine scheduling for learning rate, even within a single eopch. No idea either why I get this issue.

**Comment by RobertHua96** [comment]
Might anyone have any updates on this? I'm about to start pre training a ViT using Dino but now I have doubts...

**Comment by robbohua** [comment]
Might just be the case our datasets need to be much bigger: https://github.com/facebookresearch/dino/issues/196

**Comment by RobertHua96** [comment]
Might be the case our datasets need to be much larger: https://github.com/facebookresearch/dino/issues/196
